providers/mimetypes.rb

```diff
@@ -0,0 +1,68 @@
+#
+## Cookbook Name:: gecos-ws-mgmt
+## Provider:: pdf_viewer
+##
+## Copyright 2013, Junta de Andalucia
+## http://www.juntadeandalucia.es/
+##
+## All rights reserved - EUPL License V 1.1
+## http://www.osor.eu/eupl
+##
+
+require 'etc'
+require 'chef/mixin/shell_out'
+include Chef::Mixin::ShellOut
+
+action :setup do
+	begin
+	  if new_resource.support_os.include?($gecos_os)
+    	package "xdg-utils" do
+       action :nothing
+      end.run_action(:install)
+
+      users = new_resource.users
+
+      users.each_key do |user_key|
+        nameuser = user_key
+        username = nameuser.gsub('###','.')
+        user = users[user_key]
+				gid = Etc.getpwnam(username).gid
+				xdg_code = ''
+
+				directory "/home/#{username}/.local/share/applications" do
+					owner username
+					group gid
+					recursive true
+				end
+
+				Chef::Log.debug("mimetypes.rb - Users: #{user}")
+
+				user.mimetyperelationship.each do |assoc|
```

Comment:
Buenas, Alfonso:

No he encontrado ese comando "all". Este es el código que funciona:

           # only_if: checking desktop file exists and not already
associated with desktop file
            execute "xdg-mime default #{desktopfile} #{mime}" do
              action :run
              user username
              group gid
              environment ({'HOME' => "/home/#{username}", 'USER' =>
"#{username}"})
              command "xdg-mime default #{desktopfile} #{mime}"
              only_if %Q{[ -f /usr/share/applications/#{desktopfile} ] && [
"x$(xdg-mime query default #{mime})" != "x#{desktopfile}" ]},
:user=>username,:environment => {'HOME' => "/home/#{username}", 'USER' =>
"#{username}"}
           end

Antes de subirlo, algún comentario?

Saludos.

2016-08-02 15:01 GMT+02:00 Alfonso E.M. <notifications@github.com>:

> In providers/mimetypes.rb
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145#discussion_r73149726>
> :
>
> > +      users.each_key do |user_key|
> > +        nameuser = user_key
> > +        username = nameuser.gsub('###','.')
> > +        user = users[user_key]
> > +				gid = Etc.getpwnam(username).gid
> > +				xdg_code = ''
> > +
> > +				directory "/home/#{username}/.local/share/applications" do
> > +					owner username
> > +					group gid
> > +					recursive true
> > +				end
> > +
> > +				Chef::Log.debug("mimetypes.rb - Users: #{user}")
> > +
> > +				user.mimetyperelationship.each do |assoc|
>
> Exacto, salvo que encuentres una manera de mirar todos los tipos mime
> asignados con un solo comando, y luego hagas las comparaciones en la receta.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145/files/c25d460433d1f0d07780a75a6172e8a068ae51dd#r73149726>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATHDTF2fo7qins1F5CLHBxQNst1K3iHDks5qbz-TgaJpZM4JRrKY>
> .
>